### PR TITLE
test: extend the fiber.info() backtraces test

### LINKED ISSUE #7535: Test backtraces for fiber.info()
Looks like enabling stack protector may (or may not) break backtraces for fiber.info().
Need to add a test case to cover this.

### PR DESCRIPTION
**Pull request by Gumix**
**cmake: normalize ENABLE_BACKTRACE option**
CMake accepts the following case-insensitive values as true: 1, ON, YES,
TRUE, Y, or a non-zero number (including floating point numbers). This
complicates the parsing of ENABLE_BACKTRACE in `tarantool.build.options`.
Fix this by defining it to TRUE for any true value.

**test: extend the fiber.info() backtraces test**
Test that an expected Lua function can be found in one of frames.
C function is already covered by this test.

Closes #7535

### CODE REVIEW

File: test/app-luatest/fiber_parent_backtrace_test.lua [hunk 1]
```diff
@@ -29,13 +29,16 @@ g.test_fiber_parent_backtrace = function()
```
Comment (Totktonada) [inline test/app-luatest/fiber_parent_backtrace_test.lua:29]: I had an idea to run this test against my system tarantool on Gentoo and the test surprisingly skipped:

```lua
local _, _, enable_bt = string.find(tarantool.build.options,
                                    "-DENABLE_BACKTRACE=(%a+)")
local bt_enabled = enable_bt == "ON" or enable_bt == "TRUE"
t.skip_if(not bt_enabled, "requires backtrace feature")
```

The reason is the following:

```yaml
# tarantool> require('tarantool').build.options
---
- cmake . -DCMAKE_INSTALL_PREFIX=/usr -DENABLE_BACKTRACE=yes
...
```

The `yes` value is from https://github.com/tarantool/gentoo-overlay/blob/d76587fe613afad7fd00a0d7ac4e93fdfe95d8c7/dev-db/tarantool/tarantool-9999.ebuild#L172. We should either accept all [truthly values](https://cmake.org/cmake/help/v3.24/command/if.html#basic-expressions) (in lower/upper/mixed case) or normalize them for `require('tarantool').build.options` at build stage.

A strage thing, I can't just run `luatest -v -p <..foo..>`:

```sh
$ luatest -v -p fiber_parent_backtrace
/usr/share/tarantool/luatest/loader.lua:43: module 'test.app-luatest.http_client_test' not found:
<...>
```

So I run the test from the interactive console this way:

```lua
tarantool> t = require('luatest')
tarantool> dofile('test/app-luatest/fiber_parent_backtrace_test.lua')
tarantool> g = t.groups.x
tarantool> g.before_all()
tarantool> g.test_fiber_parent_backtrace()
---
- error: {'status': 'fail', 'class': 'LuatestError', 'message': 'test/app-luatest/fiber_parent_backtrace_test.lua:67:
      expected: 2, actual: 0'}
...
```

After the following path:

```diff
diff --git a/test/app-luatest/fiber_parent_backtrace_test.lua b/test/app-luatest/fiber_parent_backtrace_test.lua
index 1034e4121..9afbaf6fe 100644
--- a/test/app-luatest/fiber_parent_backtrace_test.lua
+++ b/test/app-luatest/fiber_parent_backtrace_test.lua
@@ -1,7 +1,7 @@
 local server = require('test.luatest_helpers.server')
 local t = require('luatest')
 
-local g = t.group()
+local g = t.group('x') -- to run from the interactive console
 
 g.before_all = function()
     g.dflt = server:new({alias = 'dflt'})
@@ -12,7 +12,7 @@ g.before_all = function()
 
         local _, _, enable_bt = string.find(tarantool.build.options,
                                             "-DENABLE_BACKTRACE=(%a+)")
-        local bt_enabled = enable_bt == "ON" or enable_bt == "TRUE"
+        local bt_enabled = enable_bt == "ON" or enable_bt == "TRUE" or enable_bt == 'yes'
         t.skip_if(not bt_enabled, "requires backtrace feature")
     end)
 end
```

To sum up:

* It worth to generalize the build option check somehow.
* The test actually works: I have [broken backtraces](https://github.com/tarantool/tarantool/issues/5326#issuecomment-1170620184) on the system tarantool (don't know exact reason, but likely it is due to the splitdebug feature) and the test able to catch the fail.
* I'm sad that something is broken about ability to run tests with luatest against system tarantool.
* Strictly speaking, everything said above is not about your pull request :)

P.S. The backtrace is the following for this setup:

```yaml
---
- C: '#1  0x557bb132a9bf in tarantool_lua_slab_cache+7681'
- C: '#2  0x557bb132aaca in tarantool_lua_slab_cache+7948'
- C: '#3  0x557bb132ab0a in tarantool_lua_slab_cache+8012'
- C: '#4  0x557bb135e06a in cord_slab_cache+951'
- C: '#5  0x557bb132b0cc in tarantool_lua_slab_cache+9486'
- L: '#6  info in =[C]:-1'
- L: '#7  some_lua_fn_name in @test/app-luatest/fiber_parent_backtrace_test.lua:33'
- L: '#8  (unnamed) in @test/app-luatest/fiber_parent_backtrace_test.lua:41'
- C: '#9  0x557bb13b254f in lua_pcall+239'
- C: '#10 0x557bb1331a82 in luaT_call+41'
- C: '#11 0x557bb132b1cf in tarantool_lua_slab_cache+9745'
- C: '#12 0x557bb10f200a in ??+0'
- C: '#13 0x557bb135ad04 in fiber_self+769'
- C: '#14 0x557bb14e1a10 in ibuf_reserve_slow+137147'
- L: '#15 create in =[C]:-1'
- L: '#16 (unnamed) in @test/app-luatest/fiber_parent_backtrace_test.lua:44'
- C: '#17 0x557bb13b254f in lua_pcall+239'
- C: '#18 0x557bb1331a82 in luaT_call+41'
- C: '#19 0x557bb132b1cf in tarantool_lua_slab_cache+9745'
- C: '#20 0x557bb10f200a in ??+0'
- C: '#21 0x557bb135ad04 in fiber_self+769'
- C: '#22 0x557bb14e1a10 in ibuf_reserve_slow+137147'
...
```

**Review by Totktonada — APPROVED** [review]
The code seems logical for me, the test passed for tarantool build locally and failed for a tarantool with broken backtraces. LGTM.

File: test/app-luatest/fiber_parent_backtrace_test.lua (hunk 1 above)
Comment (Gumix) [reply test/app-luatest/fiber_parent_backtrace_test.lua:28]: > or normalize them for `require('tarantool').build.options` at build stage

Done.

File: CMakeLists.txt [hunk 2]
```diff
@@ -712,7 +712,11 @@ endif()
 # tarantool info summary (used in server version output)
 #
 set(TARANTOOL_OPTIONS "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}")
-set(TARANTOOL_OPTIONS "${TARANTOOL_OPTIONS} -DENABLE_BACKTRACE=${ENABLE_BACKTRACE}")
+if(ENABLE_BACKTRACE)
+    set(TARANTOOL_OPTIONS "${TARANTOOL_OPTIONS} -DENABLE_BACKTRACE=TRUE")
+else()
+    set(TARANTOOL_OPTIONS "${TARANTOOL_OPTIONS} -DENABLE_BACKTRACE=FALSE")
+endif()
```
Comment (Totktonada) [inline CMakeLists.txt:719]: I looked over tarantool and its submodules (curl, luajit, libyaml, zstd, nghttp2) and it seems that ON/OFF is usually used in `option()` declarations. So I guess it is a kind of unofficial canonical form for boolean options.

Feel free to ignore it is just my guess.

**Review by Totktonada — APPROVED** [review]

**Review by locker — APPROVED** [review]

**Pull request merged**